enum.rb

```diff
@@ -0,0 +1,134 @@
+# frozen_string_literal: true
+
+module Enumerable
+  def my_each
+    return to_enum unless block_given?
+
+    i = 0
+    if block_given?
+      while i < size
+        yield(self[i])
+        i += 1
+      end
+    end
+  end
+
+  def my_each_with_index
+    return to_enum unless block_given?
+
+    i = 0
+    if block_given?
+      while i < size
+        yield(self[i], i)
+        i += 1
+      end
+    end
+  end
+
+  def my_select
+    return to_enum unless block_given?
+
+    i = 0
+    new_array = []
+    while i < size
+      new_array << self[i] if yield(self[i]) == true
+      i += 1
+    end
+    new_array
+  end
+
+  def my_all?(arg = nil)
+    if block_given?
+      my_each { |x| return false unless yield(x) }
+    elsif arg.class == Class
+      my_each { |x| return false unless x.class == arg }
+    elsif arg.class == Regexp
+      my_each { |x| return false if (x =~ arg).nil? }
+    elsif !arg.nil?
+      my_each { |x| return false unless x == arg }
+    elsif arg.nil?
+      my_each { |x| return false unless x }
+    end
+    true
+  end
+
+  def my_any?(arg = nil)
+    if block_given?
+      my_each { |x| return false unless yield(x) }
+    elsif arg.class == Class
+      my_each { |x| return false unless x.class == arg }
+    elsif arg.class == Regexp
+      my_each { |x| return false if (x =~ arg).nil? }
+    elsif arg.nil?
+      my_each { |x| return false unless x }
+    else
+      my_each { |x| return true if x == arg }
+    end
+    true
+  end
+
+  def my_none?(arg = nil)
```

Comment:
`my_none` is simply opposite of `my_any` So once you've got `my_any` working you'd simply have to negate `my_any`

Reply:
You'd need to also pass a second &block parameter along with pattern in `my_none` and pass it to your negate `my_any` to handle blocks

`def my_none?(pattern = nil, &block)
    !my_any?(pattern, &block)
  end`

Reply:
NOTE: This is not necessary or original `my_any` but for `my_none`